freqtrade/strategy/hyper.py

```diff
@@ -0,0 +1,204 @@
+"""
+IHyperStrategy interface, hyperoptable Parameter class.
+This module defines a base class for auto-hyperoptable strategies.
+"""
+import logging
+from abc import ABC, abstractmethod
+from contextlib import suppress
+from typing import Any, Iterator, Optional, Sequence, Tuple, Union
+
+
+with suppress(ImportError):
+    from skopt.space import Integer, Real, Categorical
+
+from freqtrade.exceptions import OperationalException
+
+
+logger = logging.getLogger(__name__)
+
+
+class BaseParameter(ABC):
+    """
+    Defines a parameter that can be optimized by hyperopt.
+    """
+    category: Optional[str]
+    default: Any
+    value: Any
+    opt_range: Sequence[Any]
+
+    def __init__(self, *, opt_range: Sequence[Any], default: Any, space: Optional[str] = None,
+                 enabled: bool = True, **kwargs):
```

Comment:
I'm not yet sure about the meaning of `enabled` yet (i actually only noticed it a few minutes ago)

at first, it seems good that a parameter can be excluded from hyperopt (so the paramter remains available, but is not hyperopted)

However, it's also got the side-effect that it'll not load the parameter from the params dictionary, but instead use the default (which was surprising to me)

Reply:
as said, i'm not sure about it yet - maybe it should be split into 2 settings?

on the one hand, manually tuning is great (prevent hyperopt for this parameter) - but i found it surprising that the default will be used in that case, not the value in buy_params (which is usually the result of an earlier hyperopt run)...

Reply:
Hmm my email response did not get into this particular thread. Anyway yeah i was thinking about two options as well. `optimize` + `load` maybe? Both would be enabled by default and user may achieve any kind of desired combination (load from config but do not optimize, optimize but do not load (probably least useful), load and optimize (default), neither load nor optimize (for manual tuning))

Reply:
yeah, i noticed that (but intentionally answered here again, not below to keep it "kindof" organized).

Your idea that sounds sensible / reasonable 👍